code/datums/sound_player.dm

```diff
@@ -190,6 +193,11 @@ datum/sound_token/proc/Mute()
 	PrivUpdateListeners()
 
 datum/sound_token/proc/PrivAddListener(var/atom/listener)
+	if(preference) // Checking if client want to hear it
```

Comment:
````
	if(preference)
		var/mob/M = listener
		if(istype(M))
			if(M.get_preference_value(preference) != GLOB.PREF_YES)
````
should really be in a proc rather than copypasted here and in `PrivUpdateListener()`.